File: RecyclerViewJava/app/src/main/java/com/devwilly/recyclerview_java/wrapper/FoodWrapper.java
package com.devwilly.recyclerview_java.wrapper;

/**
 * Created by Willy on 15/01/2018.
 */

public class FoodWrapper implements IEattingType {

    enum FoodCookingMethod {
        SAUTE, STEW, BROIL, BAKE, FRY, DEEPFRY
    }

    private String mName;
    private String mPrice;
    private FoodCookingMethod mCookingMethod;

    public FoodWrapper(String name, String price, FoodCookingMethod cookingMethod) {
        this.mName = name;
        this.mPrice = price;
        this.mCookingMethod = cookingMethod;
    }

    public FoodCookingMethod getCookingMethod() {
        return mCookingMethod;
    }

    @Override
    public String getName() {
        return mName;
    }

    @Override
    public String getPrice() {
        return mPrice;
    }

    @Override
    public int getItemType() {
        return IEattingType.FOOD;
    }
}

File: RecyclerView-Kotlin/app/src/main/java/com/devwilly/recyclerviewkotlin/wrapper/FoodWrapper.kt
package com.devwilly.recyclerviewkotlin.wrapper

/**
 * Created by Willy on 15/01/2018.
 */
class FoodWrapper : IEatingWrapper {

    private lateinit var mName: String
    private lateinit var mDescription: String
    private lateinit var mPrice: String

    fun FoodWrapper(name: String, description: String, price: String) {
        this.mName = name
        this.mDescription = description
        this.mPrice = price
    }
    override fun getName(): String {
        return mName
    }

    override fun getDescription(): String {
        return mDescription
    }

    override fun getPrice(): String {
        return mPrice
    }

    override fun getItemType(): Int {
        return IEatingWrapper.FOOD
    }
}